src/NuGet.Clients/NuGet.PackageManagement.UI/Utility/PackageSourceMappingUtility.cs

```diff
@@ -91,5 +92,14 @@ internal static class PackageSourceMappingUtility
 
             mappingProvider.SavePackageSourceMappings(newAndExistingPackageSourceMappingItems);
         }
+
+        internal static string? GetNewSourceMappingSourceName(PackageSourceMapping packageSourceMapping, PackageSourceMoniker activePackageSourceMoniker)
+        {
+            string? sourceMappingSourceName = packageSourceMapping.IsEnabled
+                && activePackageSourceMoniker.IsAggregateSource == false
+                ? activePackageSourceMoniker.PackageSourceNames.First() : null;
```

Comment:
If I understand this logic correctly, if a package is found in multiple package sources, then we are adding a source mapping entry for the first source, correct?

Reply:
This algorithm needs work based on my Todo section.  Basically, think of this as taking your PM UI package source drop-down selected source. Unless it's "All", then I'm returning null. The First() currently is meaningless since there's only 1 selected source possible